main.tf

```diff
@@ -103,25 +116,36 @@ resource "azurerm_role_assignment" "key_vault_rbac_current_user" {
   scope                = azurerm_key_vault.KeyVault.id
   role_definition_name = "Key Vault Administrator"
   principal_id         = data.azurerm_client_config.current.object_id
+
+  depends_on = [azurerm_key_vault.KeyVault]
 }
 
 // Create theKey Vault Secrets for later use
 
 resource "azurerm_key_vault_secret" "cd_azure_devops_pat" {
   name         = "cd-azure-devops-pat"
-  value        = ""
+  value        = var.cd_azure_devops_pat_value
   key_vault_id = azurerm_key_vault.KeyVault.id
+
+  depends_on = [azurerm_role_assignment.key_vault_rbac_current_user]
 }
 
 resource "azurerm_key_vault_secret" "cd_github_pat" {
   name         = "cd-github-pat"
-  value        = ""
+  value        = var.cd_github_pat_value
   key_vault_id = azurerm_key_vault.KeyVault.id
+
+  depends_on = [azurerm_role_assignment.key_vault_rbac_current_user]
 }
 
 //Create the provider file for the next module
 
 resource "local_file" "terraform_providers" {
   filename = "${path.root}/../2.0.0.CDMonitoring_Prerequisite_Deployment/providers.tf"
   content  = local.terraform_providers
-}
\ No newline at end of file
+}
+
+resource "local_file" "terraform_tfvars" {
+  filename = "${path.root}/../2.0.0.CDMonitoring_Prerequisite_Deployment/terraform.tfvars"
```

Comment:
And same again for this?

Reply:
It wont need to reference the templates folder as the location is provided by content which in turn is sourced from the terraform_providers and terraform_tfvars locals respectively